Update C++ documentations (#21925)
DataDog/documentation: content/en/tracing/trace_collection/library_config/cpp.md

`DD_TRACE_SAMPLING_RULES` 
: **Since**: v0.1.0 <br>
**Default**: `null` <br>
**Examples:**<br>
Set the sample rate to 20%: `[{"sample_rate": 0.2}]` <br>
Set the span sample rate to 50% for the service `my-service` and operation name `http.request`, up to 50 traces per second: `'[{"service": "my-service", "name": "http.request", "sample_rate":0.5, "max_per_second": 50}]'` <br><br>
A JSON array of objects. Each object must have a `sample_rate`, and the `name` and `service` fields are optional. The `sample_rate` value must be between 0.0 and 1.0 (inclusive). Rules are applied in configured order to determine the trace's sample rate. <br>
For more information, see [Ingestion Mechanisms][2].<br>

`DD_SPAN_SAMPLING_RULES`
: **Version**: v0.1.0 <br>
**Default**: `null`<br>
A JSON array of objects. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between `0.0` and `1.0` (inclusive).

`DD_SPAN_SAMPLING_RULES_FILE`
: **Since**: 0.1.0 <br>
Points to a JSON file that contains the span sampling rules. See `DD_SPAN_SAMPLING_RULES` for the rule format.

`DD_PROPAGATION_STYLE`
: **Since**: 0.1.0 <br>

> Comma separated list of propagation style(s) to use when extracting and injecting tracing context. <br>

Reviewer: ```suggestion
Comma separated list of propagation styles to use when extracting and injecting tracing context. <br>
```

When multiple values are given, the order of matching is based on the order of values.

`DD_TRACE_PROPAGATION_STYLE_INJECT` 
: **Since**: v0.1.6 <br>
**Default**: `datadog,tracecontext` <br>
**Accepted values**: `datadog`, `tracecontext`, `b3` <br>
Comma separated list of propagation styles to use when injecting tracing context.
When multiple values are given, the order of matching is based on the order of values.

`DD_TRACE_PROPAGATION_STYLE_EXTRACT` 
: **Since**: v0.1.6 <br>
**Default**: `datadog,tracecontext` <br>
**Accepted values**: `datadog`, `tracecontext`, `b3` <br>

> Comma separated list of propagation style to use when extracting tracing context. 

Reviewer: ```suggestion
Comma separated list of propagation styles to use when extracting tracing context. 
```

When multiple values are given, the order of matching is based on the order of values.

`DD_TRACE_ENABLED`
: **Since**: 0.1.0 <br>
**Default**: `true` <br>
Submit or not traces to the Datadog Agent. <br>
When `false`, the library stop sending traces to the Datadog Agent. However, the library continues to generate traces, report telemetry and poll for remote configuration updates.

`DD_TRACE_REPORT_HOSTNAME`
: **Since**: 0.1.0 <br>
**Default**: `false` <br>
Adds the `hostname` tag with the result of `gethostname`.

`DD_TRACE_STARTUP_LOGS`
: **Since**: 0.1.0 <br>
**Default**: `true` <br>
Log the tracer configuration once the tracer is fully initialized. <br>

`DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED`
: **Since**: 0.1.6 <br>
**Default**: `true` <br>
If `true`, the tracer will generate 128-bit trace IDs. <br>
If `false`, the tracer will generate legacy 64-bit trace IDs.

`DD_INSTRUMENTATION_TELEMETRY_ENABLED`
: **Since**: 0.1.12 <br>
**Default**: `true` <br>

> Datadog may collect [environmental and diagnostic information about your system][4] to improve the product. When `false`, telemetry data are not be collected.

Reviewer: ```suggestion
Datadog may collect [environmental and diagnostic information about your system][4] to improve the product. When `false`, telemetry data are not collected.
```

`DD_REMOTE_CONFIGURATION_ENABLED`
: **Since**: 0.2.0 <br>
**Default**: `true` <br>
Enable the capability that allows to remotely configure and change the behavior of the tracer. <br>
When `false` this feature is disabled. <br>

> For more information, see [Remote Configuration][5]

Reviewer: ```suggestion
For more information, see [Remote Configuration][5].
```
